src/filter/TargetingFilter.ts

```diff
@@ -130,14 +129,39 @@ function constructAudienceContextId(featureName: string, userId: string | undefi
     return contextId
 }
 
-function stringToUint32(str: string): number {
-    // Create a SHA-256 hash of the string
-    const hash = createHash("sha256").update(str).digest();
+async function stringToUint32(str: string): Promise<number> {
+    let crypto;
 
-    // Get the first 4 bytes of the hash
-    const first4Bytes = hash.subarray(0, 4);
+    // Check for browser environment
+    if (typeof window !== "undefined" && window.crypto && window.crypto.subtle) {
+        crypto = window.crypto;
+    }
+    // Check for Node.js environment
+    else if (typeof global !== "undefined" && global.crypto) {
+        crypto = global.crypto;
+    }
+    // Fallback to native Node.js crypto module
+    else {
+        try {
+            crypto = require("crypto");
+        } catch (error) {
+            console.error("Failed to load the crypto module:", error.message);
+            throw error;
+        }
+    }
 
-    // Convert the 4 bytes to a uint32 with little-endian encoding
-    const uint32 = first4Bytes.readUInt32LE(0);
-    return uint32;
+    // In the browser, use crypto.subtle.digest
+    if (crypto.subtle) {
+        const data = new TextEncoder().encode(str);
+        const hashBuffer = await crypto.subtle.digest("SHA-256", data);

```

Comment:
crypto.subtle.digest is an async, won't it be inconvenient?

Reply:
I don't know why a method for sha256 was designed to be async. Yes, it is inconvenient. But our featureManager.IsEnabled is async and targetingFilter.evaluate will only be called by feature manager. So it will be fine.

Reply:
yes, there would be no problem here in FM lib. That's why we designed return value of `IFeatureFilter.evaluate` as boolean or Promise<boolean> to get best flexibility/compatibility.
Never mind, just noticed the difference between browser crypto api and nodejs native crypto module. Safe to go.